core/aws_ddk_core/stages/dms_s3_to_s3.py

```diff
@@ -0,0 +1,148 @@
+            should interpret the data from Amazon S3.
+            See: https://docs.aws.amazon.com/dms/latest/userguide/CHAP_Source.S3.html#CHAP_Source.S3.ExternalTableDef
+        table_mappings: str
+            The table mappings for the task, in JSON format.
+            For more information see:
+            https://docs.aws.amazon.com/dms/latest/userguide/CHAP_Tasks.CustomizingTasks.TableMapping.html
+        """
+        super().__init__(scope, id)
+
+        self._event_source: str = f"{id}-event-source"
+        self._event_detail_type: str = f"{id}-event-type"
+
+        DMSFactory.replication_task(
+            self,
+            id=f"{id}-dms-replication-task",
+            environment_id=environment_id,
+            replication_instance_arn=DMSFactory.replication_instance(
+                self,
+                id=f"{id}-dms-replication-instance",
+                environment_id=environment_id,
+                replication_instance_class=replication_instance_class,
+            ).ref,
+            source_endpoint_arn=DMSFactory.endpoint(
+                self,
+                id=f"{id}-source-dms-endpoint",
+                environment_id=environment_id,
+                endpoint_type="source",
+                engine_name="s3",
+                s3_settings=DMSFactory.endpoint_settings_s3(
+                    self,
+                    id=f"{id}-source-s3-settings",
+                    environment_id=environment_id,
+                    bucket_name=source_bucket.bucket_name,
+                    external_table_definition=external_table_definition,
+                ),
+            ).ref,
+            target_endpoint_arn=DMSFactory.endpoint(
+                self,
+                id=f"{id}-target-dms-endpoint",
+                environment_id=environment_id,
+                endpoint_type="target",
+                engine_name="s3",
+                s3_settings=DMSFactory.endpoint_settings_s3(
+                    self,
+                    id=f"{id}-target-s3-settings",
+                    environment_id=environment_id,
+                    bucket_name=target_bucket.bucket_name,
+                ),
+            ).ref,
+            table_mappings=json.dumps(
+                {
+                    "rules": [
+                        {
+                            "rule-type": "selection",
+                            "rule-id": "1",
+                            "rule-name": "1",
+                            "object-locator": {"schema-name": "%", "table-name": "%"},
+                            "rule-action": "include",
+                            "filters": [],
+                        }
+                    ]
+                },
+            )
+            if not table_mappings
+            else table_mappings,
+        )
+
+        event_detail: Dict[str, Any] = {
+            "bucket": {
+                "name": [target_bucket.bucket_name],
```

Comment:
Should we add prefix here as well?

Reply:
https://github.com/awslabs/aws-ddk/pull/105/commits/4c1719f9a066e259d68bca6956e6d47240494db6